posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -378,49 +379,63 @@ public class PostHog private constructor(
                 return
             }
 
-            var snapshotEvent = false
-            if (event == "\$snapshot") {
-                snapshotEvent = true
-            }
+            val personProfileMode = PersonProfileMode.fromConfig(config?.personProfiles)
+            val processPersonProfile = personProfileMode.shouldProcessProfile(event)
 
-            var groupIdentify = false
-            if (event == GROUP_IDENTIFY) {
-                groupIdentify = true
+            if (!processPersonProfile) {
+                val modifiedProperties = properties?.toMutableMap() ?: mutableMapOf()
+                modifiedProperties["\$process_person_profile"] = false
+                captureInternal(event, anonymousId, modifiedProperties, userProperties, userPropertiesSetOnce, groups)
+                return
             }
 
-            val mergedProperties =
-                buildProperties(
-                    newDistinctId,
-                    properties = properties,
-                    userProperties = userProperties,
-                    userPropertiesSetOnce = userPropertiesSetOnce,
-                    groups = groups,
-                    // only append shared props if not a snapshot event
-                    appendSharedProps = !snapshotEvent,
-                    // only append groups if not a group identify event
-                    appendGroups = !groupIdentify,
-                )
+            captureInternal(event, newDistinctId, properties, userProperties, userPropertiesSetOnce, groups)
+        } catch (e: Throwable) {
+            config?.logger?.log("Capture failed: $e.")
+        }
+    }
 
-            // sanitize the properties or fallback to the original properties
-            val sanitizedProperties = config?.propertiesSanitizer?.sanitize(mergedProperties.toMutableMap()) ?: mergedProperties
+    private fun captureInternal(
```

Comment:
I think this method should take a `processPersonProfile`, so you don't need the if condition in the L385, not an early return to avoid another call to `captureInternal`

Reply:
Or even make this flag until the `buildProperties` method, and append the `$process_person_profile` there